google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/apis/GoogleCloudApiSelectorPanelTest.java

```diff
@@ -303,6 +307,48 @@ public void getPanel_withNoAvailableBomVersions_hidesBomUi() {
     assertThat(panel.getBomComboBox().isVisible()).isFalse();
   }
 
+  @Test
+  public void getPanel_withVersionReturnedFromBomQuery_displaysVersionFromBom() {
+    // TODO (eshaul): remove once feature is released
+    when(pluginInfoService.shouldEnable(GctFeature.BOM)).thenReturn(true);
+
+    String libVersion = "9.9-alpha";
+    when(mavenService.getManagedDependencyVersion(any(), anyString()))
+        .thenReturn(Optional.of(libVersion));
+    when(mavenService.getBomVersions()).thenReturn(ImmutableList.of(newTestVersion(BOM_VERSION)));
+
+    CloudLibrary cloudLibrary = LIBRARY_1.toCloudLibrary();
+    GoogleCloudApiSelectorPanel panel =
+        new GoogleCloudApiSelectorPanel(ImmutableList.of(cloudLibrary), testFixture.getProject());
+
+    panel
+        .getDetailsPanel()
+        .setCloudLibrary(cloudLibrary, BOM_VERSION, panel.getApiManagementMap().get(cloudLibrary));
+
+    assertThat(panel.getDetailsPanel().getVersionLabel().getText())
+        .isEqualTo("Version: " + libVersion);
+  }
+
+  @Test
+  public void getPanel_withNoVersionReturnedFromBomQuery_fallsBackToAndDisplaysStaticVersion() {
+    // TODO (eshaul): remove once feature is released
+    when(pluginInfoService.shouldEnable(GctFeature.BOM)).thenReturn(true);
+
+    when(mavenService.getBomVersions()).thenReturn(ImmutableList.of(newTestVersion(BOM_VERSION)));
```

Comment:
What would happen if the BOM version list is returned empty instead - would it have the same behavior?

Reply:
good eye. this mock is a red herring: its irrelevent to this test. The part that's important is the call to `setCloudLibrary` with the passed in BOM version (in this case null).

I'll remove this mock from both of the new test cases

Reply:
done